src/test/java/tictactoe/gui/GuiMediatorTest.java

```diff
@@ -0,0 +1,87 @@
+package tictactoe.gui;
+
+import javafx.scene.Node;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import org.junit.Test;
+import org.testfx.framework.junit.ApplicationTest;
+
+import java.util.ArrayList;
+
+import static org.testfx.api.FxAssert.verifyThat;
+import static org.testfx.matcher.base.NodeMatchers.hasChildren;
+import static org.testfx.matcher.control.TextMatchers.hasText;
+
+public class GuiMediatorTest extends ApplicationTest {
+
+    private GuiMediator guiMediator;
+    private Parent rootNode;
```

Comment:
This particular test is functioning like an integration test where a robot clicks around the window, the root node is kind of the equivalent of the a web browser DOM tree, I wasn't sure how else to test it otherwise...